cmd/infra/azure/create.go

```diff
@@ -148,13 +150,31 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 	}
 	l.Info("Successfully assigned contributor role to managed identity", "name", identityID)
 
-	// Create network security group
-	securityGroupName, securityGroupID, err := createSecurityGroup(ctx, subscriptionID, resourceGroupName, o.Name, o.InfraID, o.Location, azureCreds)
-	if err != nil {
-		return nil, err
+	securityGroupID := ""
+
+	if o.NetworkSecurityGroup != "" {
+		//check that the provided network security group exists
+		securityGroupClient, err := armnetwork.NewSecurityGroupsClient(subscriptionID, azureCreds, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create security group client: %w", err)
+		}
+		nsg, err := securityGroupClient.Get(ctx, resourceGroupName, o.NetworkSecurityGroup, nil)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get network security group: %w", err)
+		}
+		securityGroupID = *nsg.ID
+		result.SecurityGroupName = o.NetworkSecurityGroup
+		l.Info("Successfully found existing network security group", "name", o.NetworkSecurityGroup)
+	} else {
```

Comment:
We don't want to create it in this case since this is BYO NSG

Reply:
This only creates the security group if  opts.NetworkSecurityGroup  is an empty string

Reply:
We still don't want to create it though. We already create a NSG in the infra code - https://github.com/openshift/hypershift/blob/964d9fe7c6111a2a154e9d85ce0005d744085055/cmd/infra/azure/create.go#L152

Reply:
Nevermind, I was thinking we were in a different file SMH